src/ccnl-pkt/include/ccnl-pkt-builder.h

```diff
@@ -72,8 +72,23 @@ ccnl_mkContentObject(struct ccnl_prefix_s *name,
                      unsigned char *payload, int paylen);
 
 struct ccnl_buf_s*
+ccnl_mkSimpleContent_scratch(unsigned char *scratch, int scratch_len,
+                             struct ccnl_prefix_s *name,
+                             unsigned char *payload, int paylen, int *payoffset);
+
+static inline struct ccnl_buf_s*
 ccnl_mkSimpleContent(struct ccnl_prefix_s *name,
-                     unsigned char *payload, int paylen, int *payoffset);
+                     unsigned char *payload, int paylen, int *payoffset)
+{
+    struct ccnl_buf_s *buf = NULL;
+
+    unsigned char *tmp = (unsigned char*) ccnl_malloc(CCNL_MAX_PACKET_SIZE);
```

Comment:
Shouldn't we add a check if the ``ccnl_malloc`` call fails? This properly has side effects (code calling ``ccnl_mkSimpleContent`` assumes that the function always returns successfully) and make it a cumbersome task.

Reply:
There a many cases where that has to be fixed.